ros/api/v0/hosts.py

```diff
@@ -92,8 +92,12 @@ def get(self):
         for row in query_results:
             system_dict = row.System.__dict__
             host = {skey: system_dict[skey] for skey in system_columns}
-            host['recommendation_count'] = len(host['rule_hit_details'])
-            host['state'] = host['rule_hit_details'][0].get('key')
+            if host['rule_hit_details']:
```

Comment:
@patilsuraj767 I do not quite understand that why do we get `host['rule_hit_details'] : None` even when there is a non-empty report that we have assigned to the `rule_hit_details` column under `systems` table and it does have the value in that record.

Reply:
I guess you need to add `rule_hit_details` key [here](https://github.com/RedHatInsights/ros-backend/pull/52/files#diff-5b6c544a7a56034be87e719ba2ee43d4078f705de9c3049ee8dc5b7f8ee09382R32) for it to be available.

Reply:
As we have two processors, it is possible that the inventory_event processor will process the report and will create the system entry under the system table and for some reason insights-engine-result processor might fail to populate the `rule_hit_details` in the system table. In this case host['rule_hit_details'] will return `None`.